rust/arrow/src/array/iterator.rs

```diff
@@ -118,7 +118,9 @@ impl<'a> std::iter::Iterator for BooleanIter<'a> {
         } else {
             let old = self.current;
             self.current += 1;
-            Some(Some(self.array.value(old)))
+            // Safety:
+            // we just checked bounds
```

Comment:
I've been wanting to change this, thanks for beating me to it :)

It's not critical, but could this clarify that the bounds check happens in the self.current_end comparison?  It seems to me this also depends on an undocumented struct invariant that self.current_end <= array.len().  Upon reading the comment, I had assumed that is_null was expected to be the bounds check, but ArrayData.is_null short circuits to false if there is no null buffer, so wouldn't always bounds check.  (this also applies to the other safety comments)

Reply:
Right, I will add that to prevent future confusion. :+1